drivers/at24cxxx/at24cxxx.c

```diff
@@ -243,6 +257,22 @@ int at24cxxx_write(const at24cxxx_t *dev, uint32_t pos, const void *data,
     return check;
 }
 
+int at24cxxx_write_page(const at24cxxx_t *dev, uint32_t page, uint32_t offset,
+                        const void *data, size_t len)
+{
+    int check;
+
+    /* write no more than to the end of the current page to prevent wrap-around */
+    size_t remaining = DEV_PAGE_SIZE - offset;
```

Comment:
This could produce an invalid `remaining` size, if a smart user enters an `offset` > `DEV_PAGE_SIZE`.
So I think either this function should normalize `page` and `offset`, or the documentation should prohibit to pass an offset that is larger then the page size. I am in favor of the latter, because  `mtd_write_page()` also performs the normalization. But then I´d also add an `assert()` here.

Reply:
I amended the documentation and added an `assert()`.